### PR TITLE
Remove support for using Sync Halibut in TentacleClient

### PR DESCRIPTION
**Pull request by sburmanoctopus**
# Background

This PR removes support for using Sync Halibut in Tentacle Client.

# Results

Tentacle Client now only support Async Halibut and Async RPC calls

Related to OctopusDeploy/Issues#8266

[sc-63376]

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered the [appropriate target version for this PR](https://octopushq.atlassian.net/wiki/spaces/RND/pages/1145896982/Which+versions+should+I+patch#Process).
- [ ] I have considered appropriate testing for my change.
  - Automated testing / Exploratory testing / Nothing required?
  - Is the build green?
- [ ] I have considered manually testing my changes on a [branch instance](https://octopushq.atlassian.net/wiki/spaces/RND/pages/1262256152/Use+a+Core+Feature+Branch+Instance) to check correctness, stability and performance on Octopus Cloud.
- [ ] I have considered safety nets to reduce any time-to-recovery for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/Support/ClientAndTentacleBuilder.cs [hunk 1]
```diff
@@ -149,12 +141,7 @@ public async Task<ClientAndTentacle> Build(CancellationToken cancellationToken)
             var serverHalibutRuntimeBuilder = new HalibutRuntimeBuilder()
                 .WithServerCertificate(Certificates.Server)
                 .WithLegacyContractSupport();
-
-            if (asyncHalibutFeature.IsEnabled())
-            {
-                serverHalibutRuntimeBuilder.WithAsyncHalibutFeatureEnabled();
```
[the remark this answers is not in the document]
Comment (sburmanoctopus) [reply source/Octopus.Tentacle.Tests.Integration/Support/ClientAndTentacleBuilder.cs:155]: The assumption of this PR was that the Halibut work would be done, and used here.

File: source/Octopus.Tentacle/Commands/RunAgentCommand.cs [hunk 2]
```diff
@@ -35,7 +35,6 @@ public class RunAgentCommand : AbstractStandardCommand
         int wait;
         bool halibutHasStarted;
         bool workspaceCleanerHasStarted;
-        bool enableAsyncHalibut;
```
[the remark this answers is not in the document]
Comment (sburmanoctopus) [reply source/Octopus.Tentacle/Commands/RunAgentCommand.cs:38]: That was the assumption

File: source/Octopus.Tentacle.Tests.Integration/Support/ClientAndTentacleBuilder.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sburmanoctopus) [reply source/Octopus.Tentacle.Tests.Integration/Support/ClientAndTentacleBuilder.cs:155]: Due to the way things worked out, I'll put this back, and do the removal in the next PR with the updated Halibut

File: source/Octopus.Tentacle/Commands/RunAgentCommand.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sburmanoctopus) [reply source/Octopus.Tentacle/Commands/RunAgentCommand.cs:38]: Ok, that assumption turned out to not be true.

But regardless, this is the way we are heading, so removing it now or later makes no difference.

**Review by nathanwoctopusdeploy — APPROVED** [review]
LGTM although I have reverted the changes to Tentacle to force it to use async Halibut as that is a significant change, will ship as soon as this PR is merged and I think needs some more manual testing before we do commit it. I think it makes sense for it to be part of https://github.com/OctopusDeploy/OctopusTentacle/pull/646

nit: It is possible that TentacleClient is given a Halibut runtime that doesn't have async halibut enabled as we still have Halibut with sync available but we know that's not going to happen as we control Server and will fail as soon as an RPC call is made.

**Pull request merged**